tests/test_config.py

```diff
@@ -86,8 +87,24 @@ def test_methods(self, config_class):
         assert hasattr(config, "from_json_file")
 
     @pytest.mark.parametrize("config_class", ALL_CONFIG_CLASSES)
-    def test_task_type(self, config_class):
-        config_class(task_type="test")
+    @pytest.mark.parametrize("valid_task_type", list(TaskType))
+    def test_valid_task_type(self, config_class, valid_task_type):
+        r"""
+        Test if all configs work correctly for all valid task types
+        """
+        config_class(task_type=valid_task_type)
+
+    @pytest.mark.parametrize("config_class", ALL_CONFIG_CLASSES)
+    @pytest.mark.parametrize("invalid_task_type", ["test"])
```

Comment:
Small nit, since we're only testing a single invalid value (which is enough for this test), let's remove the `parametrize` and just pass this invalid value directly below. Also, let's use a value that makes it absolutely clear to the reader that it's invalid, e.g. "invalid-task-type".

Reply:
Done